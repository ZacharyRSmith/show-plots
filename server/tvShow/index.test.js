/**
 * @module TvShow test suite.
 *
 * @flow
 */
const TvShow = require('.');

const imdbIDForTheWire = 'tt0306414';

describe('TvShow', () => {
  describe('async fetchEpisodeRatings()', () => {
    it('works', () => {
      const assertExpected = episodeRatings =>
        expect(episodeRatings).toMatchSnapshot();
      return TvShow.fetchEpisodeRatings({ imdbID: imdbIDForTheWire, totalSeasons: 5 })
        .then(assertExpected);
    });
  });

  describe('async fetchImdbData()', () => {
    it('works', () => {
      const assertExpected = imdbData =>
        expect(imdbData).toMatchSnapshot();
      return TvShow.fetchImdbData({ title: 'The Wire' }).then(assertExpected);
    });
  });

  describe('async fetchDetails()', () => {
    it('works', () => {
      const assertExpected = details =>
        expect(details).toMatchSnapshot();
      return TvShow.fetchDetails({ id: imdbIDForTheWire }).then(assertExpected);
    });
  });

  describe('async search()', () => {
    it('filters out unpopular shows', () => {
      const assertExpected = shows => {
        expect(shows.length).toBe(2);
        expect(shows[0]).toMatchSnapshot();
        expect(shows[0].original_name).toBe('The Wire');
      }
      return TvShow.search({ query: 'wire' }).then(assertExpected);
    });

    it('returns `[]` if provided no query', () => {
      const assertExpected = shows =>
        expect(shows).toEqual([]);
      return TvShow.search({ query: '\n \t' }).then(assertExpected);
    });
  });

  describe('async searchImdb()', () => {
    it('works', () => {
      const assertExpected = results =>
        expect(results).toMatchSnapshot();
      return TvShow.search({ query: 'wire' }).then(assertExpected);
    });
  });
});
