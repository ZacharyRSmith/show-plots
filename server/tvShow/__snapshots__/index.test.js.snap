// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShow async fetchDetails() works 1`] = `
Object {
  "movie_results": Array [],
  "person_results": Array [],
  "tv_episode_results": Array [],
  "tv_results": Array [
    Object {
      "backdrop_path": "/4hWfYN3wiOZZXC7t6B70BF9iUvk.jpg",
      "first_air_date": "2002-06-02",
      "genre_ids": Array [
        18,
        80,
      ],
      "id": 1438,
      "name": "The Wire",
      "origin_country": Array [
        "US",
      ],
      "original_language": "en",
      "original_name": "The Wire",
      "overview": "Told from the points of view of both the Baltimore homicide and narcotics detectives and their targets, the series captures a universe in which the national war on drugs has become a permanent, self-sustaining bureaucracy, and distinctions between good and evil are routinely obliterated.",
      "popularity": 20.906364,
      "poster_path": "/dg7NuKDjmS6OzuNy33qt8kSkPA1.jpg",
      "vote_average": 8.2,
      "vote_count": 443,
    },
  ],
  "tv_season_results": Array [],
}
`;

exports[`TvShow async fetchEpisodeRatings() works 1`] = `
Array [
  Object {
    "Episodes": Array [
      Object {
        "Episode": "1",
        "Released": "2002-06-02",
        "Title": "The Target",
        "imdbID": "tt0749451",
        "imdbRating": "8.0",
      },
      Object {
        "Episode": "2",
        "Released": "2002-06-09",
        "Title": "The Detail",
        "imdbID": "tt0749448",
        "imdbRating": "8.3",
      },
      Object {
        "Episode": "3",
        "Released": "2002-06-16",
        "Title": "The Buys",
        "imdbID": "tt0749446",
        "imdbRating": "8.3",
      },
      Object {
        "Episode": "4",
        "Released": "2002-06-23",
        "Title": "Old Cases",
        "imdbID": "tt0749437",
        "imdbRating": "8.4",
      },
      Object {
        "Episode": "5",
        "Released": "2002-06-30",
        "Title": "The Pager",
        "imdbID": "tt0749450",
        "imdbRating": "8.4",
      },
      Object {
        "Episode": "6",
        "Released": "2002-07-07",
        "Title": "The Wire",
        "imdbID": "tt0749452",
        "imdbRating": "8.7",
      },
      Object {
        "Episode": "7",
        "Released": "2002-07-21",
        "Title": "One Arrest",
        "imdbID": "tt0749438",
        "imdbRating": "8.6",
      },
      Object {
        "Episode": "8",
        "Released": "2002-07-28",
        "Title": "Lessons",
        "imdbID": "tt0749433",
        "imdbRating": "8.6",
      },
      Object {
        "Episode": "9",
        "Released": "2002-08-04",
        "Title": "Game Day",
        "imdbID": "tt0749429",
        "imdbRating": "8.7",
      },
      Object {
        "Episode": "10",
        "Released": "2002-08-11",
        "Title": "The Cost",
        "imdbID": "tt0749447",
        "imdbRating": "8.7",
      },
      Object {
        "Episode": "11",
        "Released": "2002-08-18",
        "Title": "The Hunt",
        "imdbID": "tt0749449",
        "imdbRating": "8.6",
      },
      Object {
        "Episode": "12",
        "Released": "2002-09-01",
        "Title": "Cleaning Up",
        "imdbID": "tt0749424",
        "imdbRating": "8.9",
      },
      Object {
        "Episode": "13",
        "Released": "2002-09-08",
        "Title": "Sentencing",
        "imdbID": "tt0749441",
        "imdbRating": "8.7",
      },
    ],
    "Response": "True",
    "Season": "1",
    "Title": "The Wire",
    "totalSeasons": "5",
  },
  Object {
    "Episodes": Array [
      Object {
        "Episode": "1",
        "Released": "2003-06-01",
        "Title": "Ebb Tide",
        "imdbID": "tt0749428",
        "imdbRating": "7.9",
      },
      Object {
        "Episode": "2",
        "Released": "2003-06-08",
        "Title": "Collateral Damage",
        "imdbID": "tt0749425",
        "imdbRating": "8.2",
      },
      Object {
        "Episode": "3",
        "Released": "2003-06-15",
        "Title": "Hot Shots",
        "imdbID": "tt0749432",
        "imdbRating": "8.4",
      },
      Object {
        "Episode": "4",
        "Released": "2003-06-22",
        "Title": "Hard Cases",
        "imdbID": "tt0749430",
        "imdbRating": "8.3",
      },
      Object {
        "Episode": "5",
        "Released": "2003-06-29",
        "Title": "Undertow",
        "imdbID": "tt0749454",
        "imdbRating": "8.3",
      },
      Object {
        "Episode": "6",
        "Released": "2003-07-06",
        "Title": "All Prologue",
        "imdbID": "tt0749419",
        "imdbRating": "8.6",
      },
      Object {
        "Episode": "7",
        "Released": "2003-07-13",
        "Title": "Backwash",
        "imdbID": "tt0749422",
        "imdbRating": "8.5",
      },
      Object {
        "Episode": "8",
        "Released": "2003-07-27",
        "Title": "Duck and Cover",
        "imdbID": "tt0749427",
        "imdbRating": "8.7",
      },
      Object {
        "Episode": "9",
        "Released": "2003-08-03",
        "Title": "Stray Rounds",
        "imdbID": "tt0749445",
        "imdbRating": "8.8",
      },
      Object {
        "Episode": "10",
        "Released": "2003-08-10",
        "Title": "Storm Warnings",
        "imdbID": "tt0749443",
        "imdbRating": "9.1",
      },
      Object {
        "Episode": "11",
        "Released": "2003-08-17",
        "Title": "Bad Dreams",
        "imdbID": "tt0749423",
        "imdbRating": "9.3",
      },
      Object {
        "Episode": "12",
        "Released": "2003-08-24",
        "Title": "Port in a Storm",
        "imdbID": "tt0749439",
        "imdbRating": "9.2",
      },
    ],
    "Response": "True",
    "Season": "2",
    "Title": "The Wire",
    "totalSeasons": "5",
  },
  Object {
    "Episodes": Array [
      Object {
        "Episode": "1",
        "Released": "2004-09-19",
        "Title": "Time After Time",
        "imdbID": "tt0749453",
        "imdbRating": "8.4",
      },
      Object {
        "Episode": "2",
        "Released": "2004-09-27",
        "Title": "All Due Respect",
        "imdbID": "tt0749418",
        "imdbRating": "8.7",
      },
      Object {
        "Episode": "3",
        "Released": "2004-10-03",
        "Title": "Dead Soldiers",
        "imdbID": "tt0749426",
        "imdbRating": "8.6",
      },
      Object {
        "Episode": "4",
        "Released": "2004-10-10",
        "Title": "Hamsterdam",
        "imdbID": "tt0749420",
        "imdbRating": "8.7",
      },
      Object {
        "Episode": "5",
        "Released": "2004-10-17",
        "Title": "Straight and True",
        "imdbID": "tt0749444",
        "imdbRating": "8.8",
      },
      Object {
        "Episode": "6",
        "Released": "2004-10-31",
        "Title": "Homecoming",
        "imdbID": "tt0749431",
        "imdbRating": "8.9",
      },
      Object {
        "Episode": "7",
        "Released": "2004-11-07",
        "Title": "Back Burners",
        "imdbID": "tt0749421",
        "imdbRating": "8.6",
      },
      Object {
        "Episode": "8",
        "Released": "2004-11-14",
        "Title": "Moral Midgetry",
        "imdbID": "tt0749436",
        "imdbRating": "9.0",
      },
      Object {
        "Episode": "9",
        "Released": "2004-11-21",
        "Title": "Slapstick",
        "imdbID": "tt0749442",
        "imdbRating": "8.9",
      },
      Object {
        "Episode": "10",
        "Released": "2004-11-28",
        "Title": "Reformation",
        "imdbID": "tt0749440",
        "imdbRating": "8.9",
      },
      Object {
        "Episode": "11",
        "Released": "2004-12-12",
        "Title": "Middle Ground",
        "imdbID": "tt0749434",
        "imdbRating": "9.6",
      },
      Object {
        "Episode": "12",
        "Released": "2004-12-19",
        "Title": "Mission Accomplished",
        "imdbID": "tt0749435",
        "imdbRating": "9.4",
      },
    ],
    "Response": "True",
    "Season": "3",
    "Title": "The Wire",
    "totalSeasons": "5",
  },
  Object {
    "Episodes": Array [
      Object {
        "Episode": "1",
        "Released": "2006-09-10",
        "Title": "Boys of Summer",
        "imdbID": "tt0763093",
        "imdbRating": "8.4",
      },
      Object {
        "Episode": "2",
        "Released": "2006-09-17",
        "Title": "Soft Eyes",
        "imdbID": "tt0763095",
        "imdbRating": "8.7",
      },
      Object {
        "Episode": "3",
        "Released": "2006-09-24",
        "Title": "Home Rooms",
        "imdbID": "tt0763096",
        "imdbRating": "8.8",
      },
      Object {
        "Episode": "4",
        "Released": "2006-10-01",
        "Title": "Refugees",
        "imdbID": "tt0763097",
        "imdbRating": "8.6",
      },
      Object {
        "Episode": "5",
        "Released": "2006-10-08",
        "Title": "Alliances",
        "imdbID": "tt0763098",
        "imdbRating": "8.8",
      },
      Object {
        "Episode": "6",
        "Released": "2006-10-15",
        "Title": "Margin of Error",
        "imdbID": "tt0763099",
        "imdbRating": "8.9",
      },
      Object {
        "Episode": "7",
        "Released": "2006-10-29",
        "Title": "Unto Others",
        "imdbID": "tt0758650",
        "imdbRating": "8.7",
      },
      Object {
        "Episode": "8",
        "Released": "2006-11-05",
        "Title": "Corner Boys",
        "imdbID": "tt0796549",
        "imdbRating": "8.8",
      },
      Object {
        "Episode": "9",
        "Released": "2006-11-12",
        "Title": "Know Your Place",
        "imdbID": "tt0763100",
        "imdbRating": "8.7",
      },
      Object {
        "Episode": "10",
        "Released": "2006-11-19",
        "Title": "Misgivings",
        "imdbID": "tt0763094",
        "imdbRating": "9.0",
      },
      Object {
        "Episode": "11",
        "Released": "2006-11-26",
        "Title": "A New Day",
        "imdbID": "tt0782660",
        "imdbRating": "9.1",
      },
      Object {
        "Episode": "12",
        "Released": "2006-12-03",
        "Title": "That's Got His Own",
        "imdbID": "tt0796551",
        "imdbRating": "9.0",
      },
      Object {
        "Episode": "13",
        "Released": "2006-12-10",
        "Title": "Final Grades",
        "imdbID": "tt0796550",
        "imdbRating": "9.1",
      },
    ],
    "Response": "True",
    "Season": "4",
    "Title": "The Wire",
    "totalSeasons": "5",
  },
  Object {
    "Episodes": Array [
      Object {
        "Episode": "1",
        "Released": "2008-01-06",
        "Title": "More with Less",
        "imdbID": "tt0977178",
        "imdbRating": "8.1",
      },
      Object {
        "Episode": "2",
        "Released": "2008-01-13",
        "Title": "Unconfirmed Reports",
        "imdbID": "tt0985516",
        "imdbRating": "8.5",
      },
      Object {
        "Episode": "3",
        "Released": "2008-01-20",
        "Title": "Not for Attribution",
        "imdbID": "tt0977180",
        "imdbRating": "8.8",
      },
      Object {
        "Episode": "4",
        "Released": "2008-01-27",
        "Title": "Transitions",
        "imdbID": "tt0977181",
        "imdbRating": "9.0",
      },
      Object {
        "Episode": "5",
        "Released": "2008-02-03",
        "Title": "React Quotes",
        "imdbID": "tt0977182",
        "imdbRating": "8.9",
      },
      Object {
        "Episode": "6",
        "Released": "2008-02-10",
        "Title": "The Dickensian Aspect",
        "imdbID": "tt0977183",
        "imdbRating": "8.6",
      },
      Object {
        "Episode": "7",
        "Released": "2008-02-17",
        "Title": "Took",
        "imdbID": "tt0977184",
        "imdbRating": "8.8",
      },
      Object {
        "Episode": "8",
        "Released": "2008-02-24",
        "Title": "Clarifications",
        "imdbID": "tt0977185",
        "imdbRating": "9.1",
      },
      Object {
        "Episode": "9",
        "Released": "2008-03-02",
        "Title": "Late Editions",
        "imdbID": "tt0977186",
        "imdbRating": "9.5",
      },
      Object {
        "Episode": "10",
        "Released": "2008-03-09",
        "Title": "-30-",
        "imdbID": "tt0977179",
        "imdbRating": "9.3",
      },
    ],
    "Response": "True",
    "Season": "5",
    "Title": "The Wire",
    "totalSeasons": "5",
  },
]
`;

exports[`TvShow async fetchImdbData() works 1`] = `
Object {
  "Actors": "Dominic West, John Doman, Deirdre Lovejoy, Wendell Pierce",
  "Awards": "Nominated for 2 Primetime Emmys. Another 14 wins & 51 nominations.",
  "Country": "USA",
  "Director": "N/A",
  "Genre": "Crime, Drama, Thriller",
  "Language": "English, Greek, Mandarin, Spanish",
  "Metascore": "N/A",
  "Plot": "Baltimore drug scene, seen through the eyes of drug dealers and law enforcement.",
  "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BNjc1NzYwODEyMV5BMl5BanBnXkFtZTcwNTcxMzU1MQ@@._V1_SX300.jpg",
  "Rated": "TV-MA",
  "Ratings": Array [
    Object {
      "Source": "Internet Movie Database",
      "Value": "9.3/10",
    },
  ],
  "Released": "02 Jun 2002",
  "Response": "True",
  "Runtime": "59 min",
  "Title": "The Wire",
  "Type": "series",
  "Writer": "David Simon",
  "Year": "2002–2008",
  "imdbID": "tt0306414",
  "imdbRating": "9.3",
  "imdbVotes": "214,306",
  "totalSeasons": "5",
}
`;

exports[`TvShow async search() filters out unpopular shows 1`] = `
Object {
  "backdrop_path": "/4hWfYN3wiOZZXC7t6B70BF9iUvk.jpg",
  "first_air_date": "2002-06-02",
  "genre_ids": Array [
    18,
    80,
  ],
  "id": 1438,
  "name": "The Wire",
  "origin_country": Array [
    "US",
  ],
  "original_language": "en",
  "original_name": "The Wire",
  "overview": "Told from the points of view of both the Baltimore homicide and narcotics detectives and their targets, the series captures a universe in which the national war on drugs has become a permanent, self-sustaining bureaucracy, and distinctions between good and evil are routinely obliterated.",
  "popularity": 20.906364,
  "poster_path": "/dg7NuKDjmS6OzuNy33qt8kSkPA1.jpg",
  "vote_average": 8.16,
  "vote_count": 443,
}
`;

exports[`TvShow async searchImdb() works 1`] = `
Array [
  Object {
    "backdrop_path": "/4hWfYN3wiOZZXC7t6B70BF9iUvk.jpg",
    "first_air_date": "2002-06-02",
    "genre_ids": Array [
      18,
      80,
    ],
    "id": 1438,
    "name": "The Wire",
    "origin_country": Array [
      "US",
    ],
    "original_language": "en",
    "original_name": "The Wire",
    "overview": "Told from the points of view of both the Baltimore homicide and narcotics detectives and their targets, the series captures a universe in which the national war on drugs has become a permanent, self-sustaining bureaucracy, and distinctions between good and evil are routinely obliterated.",
    "popularity": 20.906364,
    "poster_path": "/dg7NuKDjmS6OzuNy33qt8kSkPA1.jpg",
    "vote_average": 8.16,
    "vote_count": 443,
  },
  Object {
    "backdrop_path": "/hzP06zBNTAxWSIgB7xSqBtup0hk.jpg",
    "first_air_date": "2002-11-14",
    "genre_ids": Array [
      80,
      9648,
    ],
    "id": 6322,
    "name": "Wire in the Blood",
    "origin_country": Array [
      "GB",
    ],
    "original_language": "en",
    "original_name": "Wire in the Blood",
    "overview": "Clinical psychologist Dr Tony Hill's uncanny ability to see into the minds of murderers means he finds it difficult to distance himself from disturbing cases.",
    "popularity": 3.307529,
    "poster_path": "/gVQ6nRtLeAZzmFu7VkBwoNsl4L6.jpg",
    "vote_average": 7.34,
    "vote_count": 16,
  },
]
`;
